### PR TITLE
/pr-status 

### PR DESCRIPTION
**Pull request by PrithvijitBose**
###  Screenshots/Recordings:

https://github.com/user-attachments/assets/90a155b1-426e-4091-9b7b-aa8ec10e4679

###  Additional Notes:
- **`/pr-status` Discord Slash Command** ([`src/ghdcbot/bot.py`](file:///c:/Users/prith/OneDrive/Desktop/gitcord/Gitcord-GithubDiscordBot/src/ghdcbot/bot.py)):
  - Added ephemeral `/pr-status repo:<repo> pr_number:<number>` command with rate limiting (1 request per 3s).
  - Fetches and formats real-time PR health indicators.
- **PR Health Evaluation Engine** ([`src/ghdcbot/engine/pr_status.py`](file:///c:/Users/prith/OneDrive/Desktop/gitcord/Gitcord-GithubDiscordBot/src/ghdcbot/engine/pr_status.py)):
  - Analyzes CI/check-runs status (passed, failing, running).
  - Inspects CodeRabbit review status and unresolved review comments / threads.
  - Checks mergeability and merge conflict status.
  - Inspects mentor/reviewer approval counts and requested changes.
  - Formats user-friendly Discord status cards with actionable indicators.
- **GitHub Adapter Support** ([`src/ghdcbot/adapters/github/rest.py`](file:///c:/Users/prith/OneDrive/Desktop/gitcord/Gitcord-GithubDiscordBot/src/ghdcbot/adapters/github/rest.py)):
  - Added `get_pull_request_review_threads` GraphQL query helper to inspect resolved vs. unresolved review threads.
- **Unit Tests** ([`tests/test_pr_status.py`](file:///c:/Users/prith/OneDrive/Desktop/gitcord/Gitcord-GithubDiscordBot/tests/test_pr_status.py)):
  - Comprehensive suite of 56 test cases covering health evaluations, formatting, GraphQL thread parsing, edge cases, and error handling.

### AI Usage Disclosure:

We encourage contributors to use AI tools responsibly when creating Pull Requests. While AI can be a valuable aid, it is essential to ensure that your contributions meet the task requirements, build successfully, include relevant tests, and pass all linters. Submissions that do not meet these standards may be closed without warning to maintain the quality and integrity of the project. Please take the time to understand the changes you are proposing and their impact. AI slop is strongly discouraged and may lead to banning and blocking. Do not spam our repos with AI slop.

Check one of the checkboxes below:

- [ ] This PR does not contain AI-generated code at all.
- [x] This PR contains AI-generated code. I have read the [AI Usage Policy](https://github.com/AOSSIE-Org/Info/blob/main/AI-UsagePolicy.md) and this PR complies with this policy. I have tested the code locally and I am responsible for it.

## Checklist

- [x] My PR addresses a single issue, fixes a single bug or makes a single improvement.
- [x] My code follows the project's code style and conventions
- [x] If applicable, I have made corresponding changes or additions to the documentation
- [x] If applicable, I have made corresponding changes or additions to tests
- [x] My changes generate no new warnings or errors
- [x] I have joined the [Discord server](https://discord.gg/hjUhu33uAn) and I will share a link to this PR with the project maintainers there
- [x] I have read the [Contribution Guidelines](../CONTRIBUTING.md)
- [x] Once I submit my PR, CodeRabbit AI will automatically review it and I will address CodeRabbit's comments.
- [x] I have filled this PR template completely and carefully, and I understand that my PR may be closed without review otherwise.

## Summary by CodeRabbit

* **New Features**
  * Added a `/pr-status` command for individual pull request checks or paginated dashboards.
  * Reports CI status, mergeability, approvals, requested changes, draft state, and active review comments.
  * Classifies pull requests as safe to merge, needing attention, blocked, or draft.
  * Supports repository filtering, automatic repository detection, pagination, and configurable access controls.
* **Bug Fixes**
  * Improved handling of inaccessible or failed pull request data, including review-thread retrieval fallbacks.
* **Documentation**
  * Added `/pr-status` usage and behavior details to the README.

### CODE REVIEW

File: src/ghdcbot/bot.py [hunk 1]
```diff
@@ -928,6 +935,68 @@ async def on_message(message: discord.Message) -> None:
...
+    @app_commands.describe(
+        repo="Repository name (e.g. Gitcord-GithubDiscordBot)",
+        pr_number="Pull request number (e.g. 7)",
+    )
+    @app_commands.checks.cooldown(1, 3.0)
+    async def pr_status_cmd(
+        interaction: discord.Interaction,
+        repo: str,
+        pr_number: app_commands.Range[int, 1],
+    ) -> None:
```
[the remark this answers is not in the document]
Comment (PrithvijitBose) [reply src/ghdcbot/bot.py:956]: @CodeRabbit this resolved

**Comment by PrithvijitBose** [comment]
@shubham5080 ready to merge

**Comment by shubham5080** [comment]
Thanks @PrithvijitBose — `/pr-status` is a useful direction (CI + CodeRabbit readiness for contributors/mentors).

Requesting changes before merge:
1. Please address the open CodeRabbit findings (especially review-thread coverage / remaining majors).
2. Clarify permissions: who can run `/pr-status` (everyone vs mentor-only)? Prefer a clear gate.
3. Keep scope MVP — document rate-limit behavior and GitHub API/GraphQL usage so we don’t hit limits on large orgs.
4. Avoid unrelated churn (e.g. checklist-status noise) if possible.

Happy to re-review after those updates.

**Comment by PrithvijitBose** [comment]
@shubham5080 ready to merge

**Comment by shubham5080** [comment]
@PrithvijitBose  fix the coderabbit issue !

**Comment by shubham5080** [comment]
Resolve conflict!

**Comment by PrithvijitBose** [comment]
@shubham5080 all fixed, ready to merge
